src/wazuh_modules/wm_database.c

```diff
@@ -371,15 +368,14 @@ void sync_keys_with_wdb(keystore *keys) {
             continue;
         }
 
-        if (get_agent_group(atoi(entry->id), group, OS_SIZE_65536 + 1, NULL) < 0) {
-            *group = 0;
-        }
-
+        char* group = wdb_get_agent_group(atoi(entry->id), NULL);
```

Comment:
Here is better to use the `wdb_wmdb_sock` socket connection held by the `wm_database` module.

Reply:
You are right. Fixed!